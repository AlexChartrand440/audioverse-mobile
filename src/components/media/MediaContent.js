import React from 'react'
import PropTypes from 'prop-types'
import { View, ScrollView, Text, Image, TouchableOpacity, StyleSheet } from 'react-native'
import Swiper from 'react-native-swiper'

import I18n from '../../../locales'
import { MediaTypes } from '../../constants'
import defaultImage from '../../../assets/av-logo.png'
import Slide from '../slide/Slide'

const getSlides = (data, language) => {

  const slides = []

  // presenter
  slides.push({type: 'presenter', image: data.artwork, title: data.title, subtitle: data.artist})
  
  // description
	if (data.description) {
    slides.push({type: 'description', description: data.description})
	}
	
	// conference
	// don't show conference for books/stories
	if (data.mediaCategory != MediaTypes.book && data.conference && data.conference.length) {
		image = data.conference[0].logo != '' ? data.conference[0].photo86 : defaultImage
		slides.push({type: 'conference', image: image, title: data.conference[0].title})
	}
	
	// series
	if (data.series && data.series.length) {
		image = data.series[0].logo != '' ? data.series[0].photo86 : defaultImage
		slides.push({type: 'serie', image: image, title: data.series[0].title})
	}
	
	return slides
}

const MediaContent = ({ data, language }) => {

  const slides = getSlides(data, language)
  const recordingDate = (!data.recordingDate || data.recordingDate == '0000-00-00 00:00:00') ? '' : I18n.t('Recorded', {locale: language}) + ' ' + data.recordingDate
  let sponsor = null
  if (data.sponsor && data.sponsor.length) {
    sponsor = {...data.sponsor[0]}
    sponsor.image = sponsor.logo != '' ? sponsor.photo86 : defaultImage
  }

  return (
    <View style={styles.container}>
      {sponsor && (
        <View style={styles.content}>
          <TouchableOpacity style={styles.metadata}>
            <Image
              source={sponsor.image.toString().startsWith('http') ? {uri: sponsor.image} : sponsor.image}
              style={styles.image}
            />
            <View style={styles.info}>
              <Text style={styles.title} ellipsizeMode={'tail'} numberOfLines={1}>{sponsor.title}</Text>
              <Text style={styles.subtitle} ellipsizeMode={'tail'} numberOfLines={1}>{recordingDate}</Text>
              <Text style={styles.subtitle} ellipsizeMode={'tail'} numberOfLines={1}>{sponsor.location}</Text>
            </View>
          </TouchableOpacity>
        </View>
      )}
      <Swiper activeDotColor="#FFFFFF">
        {slides.map(slide => {
          if (slide.type === 'description') {
            return (
              <ScrollView style={styles.descriptionContainer} contentContainerStyle={styles.descriptionContentContainer} key={slide.type}>
                <Text style={styles.description}>{slide.description}</Text>
              </ScrollView>
            )
          } else {
            return <Slide key={slide.type} image={slide.image} header={slide.title} subtitle={slide.subtitle} description={slide.description} />
          }
        })}
      </Swiper>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  descriptionContainer: {
    padding: 20,
    marginBottom: 50
  },
  descriptionContentContainer: {
    flexGrow: 1,
    justifyContent : 'center'
  },
  description: {
    fontSize: 18
  },
  content: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 10
  },
  metadata: {
    flexDirection: 'row',
    alignItems: 'center',
    flex: 1,
    backgroundColor: '#CCCCCC50',
    padding: 10,
    borderRadius: 10
  },
  image: {
    width: 50,
    height: 50,
    borderRadius: 25
  },
  info: {
    flex: 1,
    paddingHorizontal: 10
  },
  title: {
    color: '#000000',
    fontSize: 16,
    fontWeight: '500'
  },
  subtitle: {
    color: '#212121',
    fontSize: 14,
    fontWeight: '300'
  },
})

MediaContent.propTypes = {
  data: PropTypes.object.isRequired,
  language: PropTypes.string.isRequired
}

export default MediaContent
