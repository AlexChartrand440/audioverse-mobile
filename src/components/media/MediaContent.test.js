import React from 'react'
import MediaContent from './MediaContent'
import defaultImage from '../../../assets/av-logo.png'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const data = { title: '', artwork: defaultImage }
  const tree = renderer.create(<MediaContent data={data} language="en" />).toJSON()
  expect(tree).toMatchSnapshot()
})
