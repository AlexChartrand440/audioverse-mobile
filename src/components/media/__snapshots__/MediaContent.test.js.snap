// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      activeDotColor="#FFFFFF"
      automaticallyAdjustContentInsets={false}
      autoplay={false}
      autoplayDirection={true}
      autoplayTimeout={2.5}
      bounces={false}
      contentContainerStyle={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      contentOffset={
        Object {
          "x": 0,
        }
      }
      disableNextButton={false}
      horizontal={true}
      index={0}
      loadMinimal={false}
      loadMinimalSize={1}
      loop={true}
      onIndexChanged={[Function]}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollsToTop={false}
      showsButtons={false}
      showsHorizontalScrollIndicator={false}
      showsPagination={true}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "height": 1334,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
                "paddingHorizontal": 10,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 128,
                  "width": 128,
                }
              }
            />
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "fontSize": 26,
                  "fontWeight": "bold",
                  "marginTop": 10,
                }
              }
            >
              
            </Text>
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "fontSize": 18,
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
